components/host-dashboard/reports-section/TransactionsOverviewSection.js

```diff
@@ -0,0 +1,245 @@
+import styled from 'styled-components';
+
+import { CollectiveType } from '../../../lib/constants/collectives';
+import { formatCurrency } from '../../../lib/currency-utils';
+import { API_V2_CONTEXT, gqlV2 } from '../../../lib/graphql/helpers';
+
+import PeriodFilter, { encodePeriod, getDateRangeFromPeriod } from '../../budget/filters/PeriodFilter';
+import CollectivePickerAsync from '../../CollectivePickerAsync';
+import Container from '../../Container';
+import { Flex } from '../../Grid';
+import Loading from '../../Loading';
+import { P, Span } from '../../Text';
+
+const FilterLabel = styled.label`
+  font-weight: 500;
+  text-transform: uppercase;
+  margin-bottom: 8px;
+  color: #4e5052;
+`;
+
+const FundAmounts = styled.div`
+  width: 100%;
+  height: 48px;
+  border-radius: 10px;
+  background-color: #29cc75;
+  border-right: 360px solid #e03f6a;
+  padding-top: 10px;
+  padding-left: 5px;
+  @media (max-width: 1025px) {
+    height: 130px;
+    background-color: #29cc75;
+    border-right: 0px;
+    border-bottom: 40px solid #e03f6a;
+  }
+`;
+
+const TotalFundsLabel = styled(Container)`
+  display: table-cell;
+  padding-left: 10px;
+  height: 26px;
+  border-radius: 5px;
+  background-color: white;
+  opacity: 80%;
+  vertical-align: middle;
+`;
+
+const Square = styled(Container)`
+  width: 8px;
+  height: 8px;
+  display: inline-block;
+  background-color: ${props => props.color};
+`;
+
+const transactionsOverviewQuery = gqlV2/* GraphQL */ `
+  query TransactionsOverviewQuery(
+    $hostSlug: String!
+    $account: [AccountReferenceInput!]
+    $dateFrom: DateTime
+    $dateTo: DateTime
+  ) {
+    host(slug: $hostSlug) {
+      id
+      legacyId
+      currency
+      contributionStats(account: $account, dateFrom: $dateFrom, dateTo: $dateTo) {
+        contributionsCount
+        oneTimeContributionsCount
+        recurringContributionsCount
+        dailyAverageIncomeAmount {
+          valueInCents
+        }
+      }
+      expenseStats(account: $account, dateFrom: $dateFrom, dateTo: $dateTo) {
+        expensesCount
+        dailyAverageAmount {
+          valueInCents
+        }
+        invoicesCount
+        reimbursementsCount
+        grantsCount
+      }
+    }
+  }
+`;
+
+const TransactionsOverviewSection = ({ hostSlug }) => {
+  const [dateFrom, setDateFrom] = useState(null);
+  const [dateTo, setDateTo] = useState(null);
+  const [collectives, setCollectives] = useState(null);
+
+  const { data, loading, refetch } = useQuery(transactionsOverviewQuery, {
+    variables: { hostSlug, dateFrom, dateTo, account: collectives },
+    context: API_V2_CONTEXT,
+  });
+  const host = data?.host;
+  const currency = host?.currency;
+  const contributionStats = host?.contributionStats;
+  const expenseStats = host?.expenseStats;
+
+  useEffect(() => {
+    // Prevent running the effect on initial mount
+    if (!dateFrom && !dateTo && !collectives) {
+      return;
+    }
+    refetch();
```

Comment:
I haven't tested, but it looks like this code will prevent re-fetching when you reset the filters:
1. Pick a date
2. Reset it
3. No refetch is done

It might not be visible here since we're not using an aggressive cache policy, but the code doesn't look future-proof.

Anyway, I don't understand why we need this `refetch` here? `useQuery` normally automatically triggers a refetch if one of the variables change.

Reply:
@Betree : I somehow thought the `refetch` was necessary but now realized that it isn't necessary at all. I've corrected the issue by removing the `useEffect` altogether. Seems to work perfectly. 🤔